src/Domain/CoordinateMaps/TimeDependent/ShapeTransitionFunctions.hpp

```diff
@@ -12,36 +12,61 @@ class er;
 namespace domain::CoordinateMaps {
 /*!
  * \ingroup CoordMapsTimeDependentGroup
- * \brief A transition function that falls off as \f$f(r) = 1 / r\f$.
+ * \brief A transition function that falls off as \f$f(r) = (ar + b) / r\f$. The
+ * coefficients \f$a\f$ and \f$b\f$ are chosen so that the map falls off
+ * linearly from 1 at `r_min` to 0 at `r_max`.
  */
 class SphereTransition : public tt::ConformsTo<protocols::TransitionFunc> {
  public:
+  SphereTransition(double r_min, double r_max)
+      : r_min_(r_min),
+        r_max_(r_max),
+        a_(-1. / (r_max_ - r_min_)),
+        b_(r_max_ / (r_max_ - r_min)) {}
+
+  explicit SphereTransition() = default;
+
   template <typename T>
-  T operator()(std::array<T, 3> coords) const {
-    return 1. / magnitude(coords);
+  T operator()(std::array<T, 3> source_coords) const {
+    return a_ + b_ / magnitude(source_coords);
+  }
+
+  double original_radius_over_radius(std::array<double, 3> target_coords,
```

Comment:
@markscheel agreed on the case where the centered target coords are at `{0, 0, 0}`. 

For the range of `r_min` and `r_max` I am less sure about. The way I understand it we have a spherical shell with `r_min` and `r_max` which we are distorting. Therefore the "forward" transition map should only accept coordinates which are between `r_min` and `r_max` (when centered). For the inverse map  the `target_coords` might well be outside `r_min` and `r_max`, if the shape map has distorted the source coordinates outside this range. So I suggest:

1. within `original_radius_over_radius` we check whether the **original radius** is between `r_min` and `r_max` and return an `std::optional` accordingly.
2. Within `operator()` and `map_over_radius` we check whether all the coordinates are between `r_min` and `r_max` and throw an error if they are not. This way we also do not have to check whether the radius is ever 0 because we do not allow `r_min` to be zero.

Reply:
Yes, you are correct.  And we should not allow r_min to be zero; there should be a different ShapeTransitionFunction for a spherical domain that includes the origin.